WcaOnRails/app/controllers/results_controller.rb

```diff
@@ -285,9 +285,9 @@ def records
     @is_until = splitted_years_param[0] == "until"
     @year = splitted_years_param[1].to_i
     if @is_only
-      @years_condition = "AND result.year = #{@year}"
+      @years_condition = @is_results ? "AND result.year = #{@year}" : "AND competition.year = #{@year}"
     elsif @is_until
-      @years_condition = "AND result.year <= #{@year}"
+      @years_condition = @is_results ? "AND result.year <= #{@year}" : "AND competition.year <= #{@year}"
```

Comment:
This is a bit tricky, I believe it should actually be the other way round. If `@is_results` is set it means we select from the `Results` table which has no year, so we can use the competition year instead. In other case `result` is just an alias for `Concise*Results` table which does in fact include the year ^^

Reply:
It turns out that was not the issue